src/compile/legend.ts

```diff
@@ -146,9 +146,9 @@ export namespace properties {
     applyMarkConfig(symbols, model,
       channel === COLOR ?
         /* For color's legend, do not set fill (when filled) or stroke (when unfilled) property from config because the the legend's `fill` or `stroke` scale should have precedence */
-        without(FILL_STROKE_CONFIG, [ filled ? 'fill' : 'stroke']) :
+        without(FILL_STROKE_CONFIG, [ filled ? 'fill' : 'stroke', 'strokeDash', 'strokeDashOffset']) :
```

Comment:
I would organize this chunk of the code a bit so that the comments still make sense.

Ideally I would do something like:
```
let config = channel === COLOR ? 
  /* For color's legend, do not set fill (when filled) or stroke (when unfilled) property from config because the the legend's `fill` or `stroke` scale should have precedence */
  ... : 
  /* For other legend, no need to omit. */
  ...;
config = without(config, ['strokeDash', 'strokeDashOffset'])

applyMarkConfig(symbols, model, config)
```